more redesign

Traf27 opened the pull request:
Design changes

updated Alerter look
updated XP progress table look
various stylesheet changes
Version notes

HTML structure changes for easier styling

added menu container
removed auto loading hardcore league, as not everyone has chars there
select loumns css fix

for day/ light mode version
corrections

Traf27 commented:
[main.zip](https://github.com/benjaminjackman/looty/files/5242716/main.zip)
zip with green/red buttons in Select Column  main.css

Traf27 commented:
**Changed hover cursor, to this, I think users will be much more inclined to check what column means if they wonederd.**  

![cursor-help](https://user-images.githubusercontent.com/5377285/95633494-10494780-0a88-11eb-9a51-6591cee72d07.png)
  
I tried to install something like : 
https://jqueryui.com/tooltip/
I have yet to succeed. Would be easier to write more detailed descriptions for Column mod buttons.

**Restyled table in XpView a bit, im thinking if move remaining xp, to its own column, and leave bar as quick look at %**  
It has row highlight on hover.

<img src="https://user-images.githubusercontent.com/5377285/95634476-4a1b4d80-0a8a-11eb-98e8-f4a5c80dd032.png" width="50%" height="">

**previous version was**  
<img src="https://user-images.githubusercontent.com/5377285/95634079-594dcb80-0a89-11eb-8a0d-c77c74ceeda3.jpg" width="50%" height="">

GGG accepted my request to use their styles/html structure, its huge relief, I would not try making tooltips from scratch. It still lot of work, but easier. They said, as long as I/we don't profit financially from it, we are good to go.

Traf27 commented:
that is current look of item tooltip in dark mode, in default it looks as it was + some minor additions. Tell me if I shall leave default look for default(old) theme, or make it also work with it?
### Old  
<img src="https://i.imgur.com/XRdRiY9.png" width="" height="">  

### New 
<img src="https://i.imgur.com/oOm7FzS.png" width="" height="">

benjaminjackman commented:
> **Changed hover cursor, to this, I think users will be much more inclined to check what column means if they wonederd.**
> (Click any image to enlarge)
> 
> ![cursor-help](https://user-images.githubusercontent.com/5377285/95633494-10494780-0a88-11eb-9a51-6591cee72d07.png)
> 
> I tried to install something like :
> https://jqueryui.com/tooltip/
> I have yet to succeed. Would be easier to write more detailed descriptions for Column mod buttons.
> 
> **Restyled table in XpView a bit, im thinking if move remaining xp, to its own column, and leave bar as quick look at %**
> It has row highlight on hover.
> 
> <img alt="" width="50%" height="" src="https://user-images.githubusercontent.com/5377285/95634476-4a1b4d80-0a8a-11eb-98e8-f4a5c80dd032.png">
> 
> **previous version was**
> <img alt="" width="50%" height="" src="https://user-images.githubusercontent.com/5377285/95634079-594dcb80-0a89-11eb-8a0d-c77c74ceeda3.jpg">
> 
> GGG accepted my request to use their styles/html structure, its huge relief, I would not try making tooltips from scratch. It still lot of work, but easier. They said, as long as I/we don't profit financially from it, we are good to go.

The XP View looks really nice, I really like the bars!

benjaminjackman commented:
Personally I prefer the older uglier version of the tooltips because they are quicker for the eye to scan since the numbers are more aligned (but not in a great way either), however it's undeniable i think that version you made with the text centered looks a lot prettier. I'm pretty ok with accepting any changes in that regard.

benjaminjackman commented:
> **Changed hover cursor, to this, I think users will be much more inclined to check what column means if they wonederd.**
> (Click any image to enlarge)

Let's hold off on the cursor with ? until the tooltips are in place. I have seen that more used for when a user is holding down a button that turns on help mode. It's say just leave it at cursor: pointer for now (which indicates a ux component is clickable).

Also you can add title="" to any html element and it will use the browsers  built in tooltip system, that's what I would probably do here atleast to get started (Though i'd still leave it with cursor:pointer)

Traf27 commented:
> > **Changed hover cursor, to this, I think users will be much more inclined to check what column means if they wonederd.**
> > (Click any image to enlarge)
> 
> Let's hold off on the cursor with ? until the tooltips are in place. I have seen that more used for when a user is holding down a button that turns on help mode. It's say just leave it at cursor: pointer for now (which indicates a ux component is clickable).
> 
> Also you can add title="" to any html element and it will use the browsers built in tooltip system, that's what I would probably do here atleast to get started (Though i'd still leave it with cursor:pointer)

Traf27 commented:
ffs second time I'm closing pull request instead of canceling sending post -_-


> Personally I prefer the older uglier version of the tooltips because they are quicker for the eye to scan since the numbers are more aligned (but not in a great way either), however it's undeniable i think that version you made with the text centered looks a lot prettier. I'm pretty ok with accepting any changes in that regard.


About tooltip - i have removed that comment with tooltip images, because I came to same conclusion as you (left aligned stats are easier to read (and I'm all about usability first, though its hard to come to term with mine inclination for "nice" design), and was about to give users choice what tooltip style to use, but I couldn't come up with a good way to do this. So im leaving it to default theme. 
Almost all other atrocities I leave to dark mode ;)

> Let's hold off on the cursor with ? until the tooltips are in place. I have seen that more used for when a user is holding down a button that turns on help mode. It's say just leave it at cursor: pointer for now (which indicates a ux component is clickable).
> 
> Also you can add title="" to any html element and it will use the browsers built in tooltip system, that's what I would probably > do here atleast to get started (Though i'd still leave it with cursor:pointer)

About ? - ok I'll revert it to pointer - I have by far less user experience than you do.
And yes in your previous ver you set the title tag to full descriptions of mods. So they were there already in place

Traf27 commented:
Welp I added option in settings to change text-align in dark mode too ;)

I'd like to make clicking on mod buttons in Select Columns pane to add filter with value ( "1" for flasks for example), because i added columns for checking if items are crafted, corrupted, mirrored, identified, veiled, synthesised. But im unsure where to put seting filter and removing filter. SelectColumnsPane.scala and LootView.scala has different scopes and i cant call setFilter() from SelectColumnsPane, any ideas ?

Traf27 commented:
I have checked this pull req on chrome, firefox, opera - everywhere looks fine.

I've also changed ver number and added release notes - if you like you can tweak something.

There was around 4000 parsing warnings from my standard leagues - i've filtered it and checked for duplicates. I'm adding here text attachment FYI. 
From those heres a list of candidates to add in near future.
```
#% increased Area Damage
#% increased Area of Effects
#% increased Attributes
#% increased Damage
#% increased Melee Damage
#% increased Attack Damage
#% increased Strength
#% increased Dexterity
#% increased Intelligence
#% increased Mine Damage
#% increased Mine Throwing Speed
#% increased Totem Damage
#% increased Totem Life
#% increased Trap Damage
#% increased Trap Throwing Speed
+# to Weapon Range
#% reduced Mana Reserved
#% chance to Dodge Attack Hits
Minions deal #% increased Damage
Minions have #% increased Area of Effect
Minions have #% increased Attack Speed
Minions have #% increased Cast Speed
Minions have #% increased maximum Life
Minions have #% increased Movement Speed
Minions have +#% to all Elemental Resistance
#% increased Totem Damage
#% increased Totem Life
Totems gain +#% to all Elemental Resistances
```
[mod-unparsed-filtered-cleaned.txt](https://github.com/benjaminjackman/looty/files/5818809/mod-unparsed-filtered-cleaned.txt)

benjaminjackman commented on looty/src/main/scala/looty/views/XpView.scala:
```diff
@@ -170,49 +203,66 @@ class XpView(implicit val pc: PoeCacher) extends View {
+    addStep("Just before you go on your run press 'Add Checkpoint", stepsListXp)
+    addStep("Do your run and don't move around any gems / items you have equipped. Gems are compared by location.", stepsListXp)
+    addStep("After you have completed your run and have changed zones press 'Add Checkpoint'", stepsListXp)
+    addStep("Goto 4, repeat until you've earned enough win that you can pay to lose.", stepsListXp)
```
:laughing:

benjaminjackman commented:
I merged this with the command line, thanks traf!
